cmake/aikidoConfig.cmake.in

```diff
@@ -0,0 +1,138 @@
+set(aikido_VERSION x.y.z)
+
+@PACKAGE_INIT@
+
+function(aikido_traverse_components output_variable)
+  set(output_components)
+
+  foreach(component ${ARGN})
+    # We should be able to elminate the <C>_VISITED property by detecting
+    # whether the <C>_FOUND property is DEFINED. Unfortunately, DEFINED checks
+    # whether or not define_property() was called, not whether the variable was
+    # set. The exact behavior seems to vary between CMake versions, so we use
+    # an extra variable instead.
+    get_property(is_visited DIRECTORY PROPERTY "aikido_${component}_VISITED")
+    set_property(DIRECTORY PROPERTY "aikido_${component}_VISITED" TRUE)
+
+    if(NOT is_visited)
+      set(component_include
+        "${CMAKE_CURRENT_LIST_DIR}/aikido_${component}Component.cmake")
+      set(target_include
+        "${CMAKE_CURRENT_LIST_DIR}/aikido_${component}Targets.cmake")
+
+      if(EXISTS "${component_include}" AND EXISTS "${target_include}")
+        include("${component_include}")
+
+        set_property(DIRECTORY PROPERTY "aikido_${component}_FOUND" TRUE)
```

Comment:
Why do we use `DIRECTORY` scope on these properties? It seems like `find_package` would normally have global scope, but I'm sure there's some bizarre convention at work here?